### DISCUSSION TITLE
Features, which used for training

### DISCUSSION THREAD

**Discussion by trover97** [microsoft/FLAML#129] [Q&A] [answered]
Hello, guys!
How can I get a list of features used for training?

**Comment by qingyun-wu** [comment, marked as answer]
Hi @trover97. If your data is in numpy ndarray format, all the features are used for training. If your data is in pandas DataFrame format, you can get the features list through ```automl_experiment._X_train_all.columns.values```  (assuming ```automl_experiment``` is a fitted automl instance from flaml). Note that if your original pandas DataFrame contains datetime features, we pre-processed such features (Ref: https://github.com/microsoft/FLAML/blob/main/flaml/data.py#L219).

Let me know if this does not answer your question.
Thanks!

**Reply by trover97** [reply]
Thanks for the reply! Yes, that's what I wanted to know)
Does FLAML do any feature selection (not pre-processing) before training if dataframe is used?

**Reply by qingyun-wu** [reply]
Currently no (though FLAML may drop certain columns of the dataframe, e,g., the columns which have only 1 unique value https://github.com/microsoft/FLAML/blob/main/flaml/data.py#L202 as part of the pre-processing step). But we are interested in incorporating feature selection in FLAML if there are such needs. Do you mind sharing your use case or data example?

Thank you!

**Reply by trover97** [reply]
Hi! Sorry for the long reply. I use your AutoML as an alternative modeling in my model development pipeline. I am training on regular tabular data. I can't share an example as it is confidential.

**Reply by qingyun-wu** [reply]
Thank you @trover97 for following up. I wonder do you find FLAML effective/useful in your application scenario? Do you find any of your needs not well satisfied? Please feel free to share if you have any comments. Your feedback is helpful for us to know and prioritize where FLAML should be improved.
